manage.py

```diff
@@ -461,6 +461,76 @@ def delete_guid_based_uploads():
+    """Delete a large proportion of records from our database by dropping all except the first two measures in each
+    subtopic, and all of the records associated with those dropped measures. This is used to keep our review app
+    databases under 10k rows, which is (as of 2019-04-01) the limit on Heroku."""
+    from application.utils import TimedExecution
+    from application.cms.models import (
+        Subtopic,
+        Measure,
+        DataSource,
+        DataSourceInMeasureVersion,
+        MeasureVersion,
+        Dimension,
+        Chart,
+        Table,
+    )
+
+    with TimedExecution("Delete 3rd measure and onward in each subtopic"):
+        try:
+            measure_ids_to_save = []
+
+            subtopics = Subtopic.query.all()
+            for subtopic in subtopics:
+                measure_ids_to_save.extend(m.id for m in subtopic.measures[:2])
+
+            measure_version_ids_to_delete = (
+                db.session.query(MeasureVersion.id).filter(~MeasureVersion.measure_id.in_(measure_ids_to_save)).all()
+            )
+
+            data_source_ids_to_delete = (
+                db.session.query(DataSourceInMeasureVersion.data_source_id)
+                .filter(DataSourceInMeasureVersion.measure_version_id.in_(measure_version_ids_to_delete))
+                .all()
+            )
+
+            dimension_guids_to_delete = (
+                db.session.query(Dimension.guid)
+                .filter(Dimension.measure_version_id.in_(measure_version_ids_to_delete))
+                .all()
+            )
+            chart_ids_to_delete = (
+                db.session.query(Dimension.chart_id)
+                .filter(Dimension.measure_version_id.in_(measure_version_ids_to_delete))
+                .all()
+            )
+            table_ids_to_delete = (
+                db.session.query(Dimension.table_id)
+                .filter(Dimension.measure_version_id.in_(measure_version_ids_to_delete))
+                .all()
+            )
+
+            db.session.query(DataSourceInMeasureVersion).filter(
+                DataSourceInMeasureVersion.data_source_id.in_(data_source_ids_to_delete)
+            ).delete(synchronize_session=False)
+            db.session.query(DataSource).filter(DataSource.id.in_(data_source_ids_to_delete)).delete(
+                synchronize_session=False
+            )
+            db.session.query(Dimension).filter(Dimension.guid.in_(dimension_guids_to_delete)).delete(
+                synchronize_session=False
+            )
+            db.session.query(Chart).filter(Chart.id.in_(chart_ids_to_delete)).delete(synchronize_session=False)
+            db.session.query(Table).filter(Table.id.in_(table_ids_to_delete)).delete(synchronize_session=False)
+            db.session.query(Measure).filter(~Measure.id.in_(measure_ids_to_save)).delete(synchronize_session=False)
+
+        except Exception as e:
+            print(e)
+
+    db.session.commit()
+
+
 # TODO: END
```

Comment:
I know you didn't add this... but this `TODO` feels not very useful.

Reply:
😬